spec-tests/src/test/java/org/candlepin/spec/pools/OneSubPoolPerStackSpecTest.java

```diff
@@ -0,0 +1,805 @@
+        return ownerApi.createPool(owner.getKey(), Pools.random(
+            createStackedDatacenterProduct(owner, ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.randomEng()), stackId))
+            .startDate(OffsetDateTime.now().minusSeconds(5L))
+            .contractNumber(DATACENTER_POOL)
+            .accountNumber("")
+            .orderNumber(""));
+    }
+
+    private PoolDTO createStackedProductDiffIdPool(OwnerDTO owner, String stackId) {
+        return ownerApi.createPool(owner.getKey(), Pools.random(
+            createRegularStackedProduct(owner, stackId))
+            .quantity(2L)
+            .startDate(OffsetDateTime.now().minusDays(3L))
+            .endDate(OffsetDateTime.now().plusDays(6L))
+            .contractNumber(REGULAR_STACKED_WITH_DIFF_STACK_ID)
+            .accountNumber("")
+            .orderNumber(""));
+    }
+
+    private void createTargetPoolAndProducts(OwnerDTO owner, String stackingId) {
+        ContentDTO ownerContent1 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ContentDTO ownerContent2 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ProductDTO ownerDerivedEngProduct = ownerProductApi.createProductByOwner(
+            owner.getKey(), Products.random());
+        ownerProductApi.addBatchContent(owner.getKey(), ownerDerivedEngProduct.getId(),
+            Map.of(ownerContent1.getId(), true, ownerContent2.getId(), true));
+
+        ProductDTO ownerDerivedSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(new AttributeDTO().name("host_limited").value("true"))));
+
+        ProductDTO ownerSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(
+                new AttributeDTO().name("host_limited").value("true"),
+                new AttributeDTO().name("stacking_id").value(stackingId),
+                new AttributeDTO().name("virt_limit").value("unlimited"))));
+
+        ownerApi.createPool(owner.getKey(), Pools.random(ownerSku)
+            .stackId(stackingId)
+            .stacked(true)
+            .startDate(OffsetDateTime.now().minusSeconds(5L))
+            .derivedProductId(ownerDerivedSku.getId())
+            .derivedProvidedProducts(Set.of(
+                new ProvidedProductDTO()
+            .productId(ownerDerivedEngProduct.getId())
+            .productName(ownerDerivedEngProduct.getName()))));
+    }
+
+    private String getAttributeValue(String name, List<AttributeDTO> attributes) {
+        List<AttributeDTO> matches = attributes.stream()
+            .filter(x -> name.equals(x.getName()))
+            .collect(Collectors.toList());
+        if (matches.size() > 0) {
+            return matches.get(0).getValue();
+        }
+        return null;
```

Comment:
Optional: You can use a ternary operator here:
`return matches.size() > 0 : matches.get(0).getValue() ? null; `